### PR TITLE
Improve Safari User-Agent patterns (#188)

### LINKED ISSUE #188: Safari section probably needs further work
I saw https://github.com/whatwg/compat/pull/187 right after I checked Safari UAs in a retail store. What a coincidence.

Here are the devices and their UA strings I checked:

iPhone SE (3rd gen, 2022):

```
Mozilla/5.0 (iPhone; CPU iPhone OS 15_4 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Mobile/15E148 Safari/604.1
```

iPad mini (6th gen, 2021):

```
Mozilla/5.0 (iPad; CPU OS 15_4 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Mobile/15E148 Safari/604.1
```

iPad Air (5th gen, 2022):

```
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Safari/605.1.15
```

iPad Pro 11" (3rd gen, 2021):

```
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Safari/605.1.15
```

So, like @karlcow said in https://github.com/whatwg/compat/issues/180#issuecomment-1079571138 , Safari on iPad has different configurations depending on their form factors; smaller devices report Mobile-ish strings, larger devices report as if they were Mac.

If I recall correctly, iPad changed to report desktop UAs with the introduction of iPadOS. I haven't checked with 12.9" Pro and 10.2". But I guess non-mini iPads report desktop-ish strings by default...?

There are other things to note:

* the OS version after `Mac OS X` on larger iPads is `10_15_6`, not `10_15_7` that is what Safari on macOS reports
* the last version token (`Safari/**`) on smaller devices is `Safari/604.1` and is not the same as `<safariVersion>`

### PR DESCRIPTION
**Pull request by miketaylr**
Fixes #188 

r? @karlcow

***
<a href="https://whatpr.org/compat/209.html" title="Last updated on Aug 1, 2022, 5:29 PM UTC (fb3082c)">Preview</a> | <a href="https://whatpr.org/compat/209/e450ea5...fb3082c.html" title="Last updated on Aug 1, 2022, 5:29 PM UTC (fb3082c)">Diff</a>

### CODE REVIEW

**Comment by miketaylr** [comment]
This depends on stuff defined in https://github.com/whatwg/compat/pull/210 - we should land that first.

File: compatibility.bs [hunk 1]
```diff
@@ -995,32 +998,34 @@ On Firefox for Android, "<code>&lt;<a>platform</a>&gt;; &lt;<a for=firefox>devic
...
+"<code>&lt;<a>majorVersion</a>&gt;.&lt;<a>minorVersion</a>&gt;</code>", where
 <code>&lt;<a>minorVersion</a>&gt;</code> is a single digit.
 
-<code>&lt;<dfn>safariPlatform</dfn>&gt;</code> decomposes to the following:
+<code>&lt;<dfn>safariPlatform</dfn>&gt;</code> <a>decomposes</a> to the following:
+
+On desktop, the <a>constant</a> "<code>Macintosh; Intel Mac OS X 10_15_7</code>".
 
-On desktop, the string "<code>Macintosh; Intel Mac OS X 10_15_7</code>".
+On larger iPad form factors, the <a>constant</a> "<code>Macintosh; Intel Mac OS X 10_15_6</code>".
```
Comment (gsnedders) [inline compatibility.bs:1026]: This recently changed (https://github.com/WebKit/WebKit/commit/2cc95bc67eb6a3ef70de894d03cc1d446c7530eb).

Comment (karlcow) [reply compatibility.bs:1026]: Thanks @gsnedders 
Could you change this @miketaylr

**Review by karlcow — CHANGES_REQUESTED** [review]
Please rebase with regards to the latest merge and make the change @gsnedders suggested.

File: compatibility.bs (hunk 1 above)
Comment (miketaylr) [reply compatibility.bs:1026]: That's great - thanks @gsnedders

**Review by karlcow — APPROVED** [review]
thanks a lot. <3 @miketaylr

**Pull request merged**